### DISCUSSION TITLE
DataGrid  Binding  ObservableCollection<MyExtendsObject>

### DISCUSSION THREAD

**Discussion by YSW33** [AvaloniaUI/Avalonia#12527] [Q&A]
Avalonia DataGrid  Binding DynamicObject invalid。
but WPF DataGrid  Binding DynamicObject is OK。

<DataGrid Name="dgTest" Margin="5"  ItemsSource="{Binding Items}" BorderThickness="2" GridLinesVisibility="Horizontal"HeadersVisibility="Column" RowHeaderWidth="50" ColumnHeaderHeight="30"/>

DataContext = new MainViewModel();

    public class MainViewModel : ViewBase
    {
        /// <summary>
        /// 绑定的数据
        /// </summary>
        ObservableCollection<MyExtendsObject> _Items = new ObservableCollection<MyExtendsObject>();

        public ObservableCollection<MyExtendsObject> Items
        {
            get { return _Items; }
            set
            {
                _Items = value;
            }
        }
    }

    public sealed class MyExtendsObject : DynamicObject
    {
        private readonly Dictionary<string, object> _properties;

        public MyExtendsObject(Dictionary<string, object> properties)
        {
            _properties = properties;
        }

        public override IEnumerable<string> GetDynamicMemberNames()
        {
            return _properties.Keys;
        }

        public override bool TryGetMember(GetMemberBinder binder, out object? result)
        {
            if (_properties.ContainsKey(binder.Name))
            {
                result = _properties[binder.Name];
                return true;
            }
            else
            {
                result = null;
                return false;
            }
        }

        public override bool TrySetMember(SetMemberBinder binder, object value)
        {
            if (_properties.ContainsKey(binder.Name))
            {
                _properties[binder.Name] = value;
                return true;
            }
            else
            {
                return false;
            }
        }
    }

**Comment by cornerbowlsoftware** [comment]
Yes, this is a major limitation.  I'd say it's a bug and I'm going to log it as such.
